addons/stock/models/stock_move.py

```diff
@@ -615,6 +615,46 @@ def action_show_details(self):
         """
         self.ensure_one()
 
+        if self.origin_returned_move_id and float_is_zero(self.quantity_done, precision_rounding=self.product_uom.rounding):
+            origin_quantity = self.origin_returned_move_id.product_uom._compute_quantity(self.origin_returned_move_id.product_uom_qty, self.product_uom, round=False)
+            if self.product_id.tracking == 'lot':
+                if float_compare(self.product_qty, origin_quantity, precision_rounding=self.product_uom.rounding) == 0:
+                    move_lines_commands = []
```

Comment:
`move_lines_commands = []` and the `self.write({'move_line_ids': move_lines_commands})` should be outside the multiple if -else.